components/normalizer/benches/composing_normalizer_nfc.rs

```diff
@@ -108,19 +112,43 @@ fn normalizer_bench_data() -> [BenchDataContent; 15] {
         content_random_words_he,
         content_random_words_de,
     ]
-    .map(|(file_name, raw_content)| BenchDataContent {
-        file_name: file_name.to_owned(),
-        nfc: nfc_normalizer.normalize(raw_content),
-        nfd: nfd_normalizer.normalize(raw_content),
-        nfkc: nfkc_normalizer.normalize(raw_content),
-        nfkd: nfkd_normalizer.normalize(raw_content),
+    .map(|(file_name, raw_content)| {
+        let nfc = &nfc_normalizer.normalize(raw_content);
+        let nfd = &nfd_normalizer.normalize(raw_content);
+        let nfkc = &nfkc_normalizer.normalize(raw_content);
+        let nfkd = &nfkd_normalizer.normalize(raw_content);
+        BenchDataContent {
+            file_name: file_name.to_owned(),
+            nfc: nfc.to_owned(),
+            nfd: nfd.to_owned(),
+            nfkc: nfkc.to_owned(),
+            nfkd: nfkd.to_owned(),
+            nfc_u16: as_utf16(nfc),
+            nfd_u16: as_utf16(nfd),
+            nfkc_u16: as_utf16(nfkc),
+            nfkd_u16: as_utf16(nfkd),
+        }
     })
 }
 
+fn as_utf16(data: &str) -> Vec<u16> {
+    let mut data_utf16 = Vec::new();
+    let mut buf = [0; 2];
+    for c in data.chars() {
+        let enc = c.encode_utf16(&mut buf);
+        data_utf16.extend_from_slice(enc);
+    }
+    data_utf16
```

Comment:
```suggestion
    s.encode_utf16().collect()
```

Also consider inlining.

Reply:
done!